src/frequenz/client/dispatch/types.py

```diff
@@ -305,6 +173,122 @@ class Dispatch:
     update_time: datetime
     """The last update time of the dispatch in UTC. Set when a dispatch is modified."""
 
+    def running(self, type_: str) -> RunningState:
+        """Check if the dispatch is currently supposed to be running.
+
+        Args:
+            type_: The type of the dispatch that should be running.
+
+        Returns:
+            RUNNING if the dispatch is running,
+            STOPPED if it is stopped,
+            DIFFERENT_TYPE if it is for a different type.
+        """
+        if self.type != type_:
+            return RunningState.DIFFERENT_TYPE
+
+        if not self.active:
+            return RunningState.STOPPED
+
+        now = datetime.now(tz=timezone.utc)
+
+        if now < self.start_time:
+            return RunningState.STOPPED
+
+        # A dispatch without duration is always running, once it started
+        if self.duration is None:
+            return RunningState.RUNNING
+
+        if until := self._until(now):
+            return RunningState.RUNNING if now < until else RunningState.STOPPED
+
+        return RunningState.STOPPED
```

Comment:
Can't we just make this a `bool` and forget about checking the `type` here? While we are at it, make it a property and rename to `is_active`. Running seems to strong here, that this library is not really running anything.

Reply:
I am very much against renaming this to `is_active`, else it can extremely easily be confused with the dispatch attribute `active`.. 

We can change it to a property `def started()`?

Reply:
> I am very much against renaming this to `is_active`, else it can extremely easily be confused with the dispatch attribute `active`..

:grimacing: 

> We can change it to a property `def started()`?

I don't like it that much, but it would match the new proposed [status](https://github.com/frequenz-floss/frequenz-api-dispatch/issues/167) name, so I guess we can go with that, and when status is implemented, we can just have a `status` property and that's it.

PS: Copilot suggested **in progress**, which I like better, but let's discuss status names in the status issue.

Reply:
No, the status property is very different from this and should be kept very separate, semantically as well.

Reply:
we could call it "should_be_running"

Reply:
or "should_target_be_running"?

Reply:
> No, the status property is very different from this and should be kept very separate, semantically as well.

Why?

Reply:
Well, the status reflects something based on reality (optimally), where as the dispatches "is_running" or however we call it is purely based on what the dispatch wants to happen.

Reply:
But the dispatch will still be the source of truth I guess, and its data will be sent to the API at some point so it is based on reality. In any case, I think it makes no sense to push for this now and the status interface is still being discussed. For the sake of advancing with this PR, I'm OK with keeping `is_running` for now, which is basically what we had before.

What's more important now IMHO is to make this a property and kill `DIFFERENT_TYPE`, if this can be done in this PR (so the client doesn't have to ever have `DIFFERENT_TYPE`) then great, otherwise we can leave it as a future improvement.